students/Tianx/Lesson4/Trigrams.py

```diff
@@ -0,0 +1,67 @@
+# ------------------------------------------#
+# !/usr/bin/env python3
+# Title: Trigrams.py
+# Desc: Trigrams – Simple Text Manipulation
+# Tian Xie, 2020-04-12, Created File
+# ------------------------------------------#
+
+import random
+
+#words = "I wish I may I wish I might".split()
+
+
+def read_in_data(filename):
+    """Read the file and remove common punctuations and lower case it.
+    """
+    with open(filename, 'r') as f:
+        words = f.read().replace("-", " ").replace("!", " ").replace(".", " ").replace(",", " ").replace('"', " ").\
+            replace("'", " ").replace("?", " ").replace("@", " ").replace(':', " ").replace(';', " ").split()
+        words = [each_word.lower() for each_word in words]
```

Comment:
nice use of comprehension here, note that you can also use `map()` function `list(map(str.lower, words))`